src/zh/dm5/src/eu/kanade/tachiyomi/extension/zh/dm5/Dm5.kt

```diff
@@ -182,8 +187,10 @@ class Dm5 : ParsedHttpSource(), ConfigurableSource {
     override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
 
     override fun imageRequest(page: Page): Request {
-        val header = headers.newBuilder().add("Referer", baseUrl).build()
-        return GET(page.imageUrl!!, header)
+        val url = page.imageUrl!!.toHttpUrl()
+        val cid = url.queryParameter("cid")!!
+        val headers = headers.newBuilder().add("Referer", "$baseUrl/m$cid").build()
```

Comment:
Not 100% sure on the order of operations, but couldn't you use `headersBuilder()` directly?
```suggestion
        val headers = headersBuilder().add("Referer", "$baseUrl/m$cid").build()
```

For reference:
https://github.com/mihonapp/mihon/blob/e8c6e3364dded0cb21df572be175cbb01adc243f/source-api/src/commonMain/kotlin/eu/kanade/tachiyomi/source/online/HttpSource.kt#L62

Reply:
In this case I'm just reusing the original code but if you dig further, `headers.newBuilder()` is actually cheaper when `headers` is initialized because it only involves [copying an `ArrayList`](https://github.com/square/okhttp/blob/8cb6ff81ab056aa9db41b0eca7de68865bf8dde6/okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/-HeadersCommon.kt#L41-L45). Calling `headersBuilder()` requires an extra pass of checking/encoding of the header values.

There's mixed usage in our codebase and I think using either is fine.